redbot/cogs/trivia/session.py

```diff
@@ -0,0 +1,229 @@
+        by `ctx.author`.
+    question_list : `dict`
+        A dict mapping questions (`str`) to answers (`list` of `str`).
+    settings : `redbot.core.config.Group`
+        Config for the trivia session.
+    scores : `collections.Counter`
+        A counter with the players as keys, and their scores as values. The
+        players are of type :py:class:`discord.Member`.
+    count : `int`
+        The number of questions which have been asked.
+    stopped : `bool`
+        Whether or not the trivia session has been stopped.
+
+    """
+
+    def __init__(self,
+                 ctx,
+                 question_list,
+                 settings):
+        self.ctx = ctx
+        self.question_list = question_list
+        self.settings = settings
+        self.scores = Counter()
+        self.count = 0
+        self.stopped = False
+        self._last_response = time.perf_counter()
+
+    async def run(self):
+        """Run the trivia session."""
+        max_score = await self.settings.max_score()
+        delay = await self.settings.delay()
+        timeout = await self.settings.timeout()
+        for question, answers in self._iter_questions():
+            self.count += 1
+            msg = "**Question number {}!**\n\n{}".format(self.count, question)
+            await self.ctx.send(msg)
+            result = await self.wait_for_answer(answers, delay, timeout)
+            if result is False:
+                break
+            if any(score >= max_score for score in self.scores.values()):
+                await self.end_game()
+                break
+            async with self.ctx.typing():
+                await asyncio.sleep(3)
+        else:
+            await self.ctx.send("There are no more questions!")
+            await self.end_game()
+
+    def _iter_questions(self):
+        questions = tuple(self.question_list.keys())
+        for _ in range(len(questions)):
+            question = random.choice(questions)
+            yield question, self.question_list.pop(question)
+
+    async def wait_for_answer(self,
+                              answers: List[str],
+                              delay: float,
+                              timeout: int):
+        """Wait for a correct answer, and then respond.
+
+        Scores are also updated in this method.
+
+        Returns False if waiting was cancelled; this is usually due to the
+        session being forcibly stopped.
+
+        Parameters
+        ----------
+        answers : `list` of `str`
+            A list of valid answers to the current question.
+        delay : float
+            How long users have to respond (in seconds).
+        timeout : int
+            How long before the session ends due to no responses (in seconds).
+
+        Returns
+        -------
+        bool
+            :code:`True` if the session wasn't interrupted.
+
+        """
+        try:
+            message = await self.ctx.bot.wait_for(
```

Comment:
I don't see anything here that would cause the issue you described. Try to debug and isolate it.